### PR TITLE
add image_data

### PR DESCRIPTION
**Pull request by BumbleFeng**
Part of https://issues.redhat.com/browse/APPSRE-3925

Design doc: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/34670

Signed-off-by: Feng Huang <fehuang@redhat.com>

### CODE REVIEW

File: schemas/openshift/terraform-resource-1.yml [hunk 1]
```diff
@@ -794,8 +788,27 @@ oneOf:
...
+            tag_name:
+              type: string
+            url:
+              type: string
+              format: uri
+            ref:
+              type: string
+              pattern: '^([0-9a-f]{40}|master|main)$'
+          required:
+          - tag_name
```
Comment (geoberle) [inline schemas/openshift/terraform-resource-1.yml:804]: i understand ref and url, but what is tag_name?

Comment (BumbleFeng) [reply schemas/openshift/terraform-resource-1.yml:804]: We assume AMI should be tagged with a tag_name and its value should be commit_sha comes from url and ref. So we can filter AMI with that tag. tag_name can be customized by tenants(e.g image builder use composer_commit)

**Comment by rporres** [comment]
Can you add documentation using `description` field to all the new schema elements?

File: schemas/openshift/terraform-resource-1.yml [hunk 2]
```diff
@@ -794,8 +788,27 @@ oneOf:
       properties:
         id:
           type: string
```
Comment (maorfr) [inline schemas/openshift/terraform-resource-1.yml:790]: as discussed, image id should be removed

**Review by maorfr — APPROVED** [review]

**Pull request merged**